src/Animals.java

```diff
@@ -0,0 +1,46 @@
+public class Animals {
+
+    private String name;
+    private static int animalsCount;
+    private static int catsCount;
+    private static int dogsCount;
+
+    public Animals(String name) {
+        this.name = name;
+        animalsCount++;
+
+        if (this instanceof Cat) {
```

Comment:
It is a bad idea to check whether the type is correct or not. 
Please answer the question: Why does this class (Animal) should know some information about cats or dogs? 
It shouldn't! 
Please remove this logic from the `Animal` class. The counters for cats and dogs should be implemented in the respective classes - `Cat` and `Dog`.